core/src/main/java/org/opensearch/sql/expression/ip/IPFunctions.java

```diff
@@ -31,75 +31,35 @@ public void register(BuiltinFunctionRepository repository) {
    * Supports both IPv4 and IPv6 addresses.
    *
-   * @param addressExprValue IP address as a string (e.g. "198.51.100.14" or
-   *     "2001:0db8::ff00:42:8329").
-   * @param rangeExprValue IP address range in CIDR notation as a string (e.g. "198.51.100.0/24" or
+   * @param addressExprValue IP address (e.g. "198.51.100.14" or "2001:0db8::ff00:42:8329").
+   * @param rangeExprValue IP address range string in CIDR notation (e.g. "198.51.100.0/24" or
    *     "2001:0db8::/32")
    * @return true if the address is in the range; otherwise false.
    * @throws SemanticCheckException if the address or range is not valid, or if they do not use the
    *     same version (IPv4 or IPv6).
    */
   private ExprValue exprCidrMatch(ExprValue addressExprValue, ExprValue rangeExprValue) {
 
-    // TODO #3145: Update to support IP address data type.
-    String addressString = addressExprValue.stringValue();
-    String rangeString = rangeExprValue.stringValue();
-
-    final IPAddressStringParameters validationOptions =
-        new IPAddressStringParameters.Builder()
-            .allowEmpty(false)
-            .setEmptyAsLoopback(false)
-            .allow_inet_aton(false)
-            .allowSingleSegment(false)
-            .toParams();
-
-    // Get and validate IP address.
-    IPAddressString address =
-        new IPAddressString(addressExprValue.stringValue(), validationOptions);
-
-    try {
-      address.validate();
-    } catch (AddressStringException e) {
-      String msg =
-          String.format(
-              "IP address '%s' is not valid. Error details: %s", addressString, e.getMessage());
-      throw new SemanticCheckException(msg, e);
-    }
-
-    // Get and validate CIDR IP address range.
-    IPAddressString range = new IPAddressString(rangeExprValue.stringValue(), validationOptions);
+    IPAddress address = addressExprValue.ipValue();
+    IPAddress range = IPUtils.toRange(rangeExprValue.stringValue());
 
-    try {
-      range.validate();
-    } catch (AddressStringException e) {
-      String msg =
-          String.format(
-              "CIDR IP address range '%s' is not valid. Error details: %s",
-              rangeString, e.getMessage());
-      throw new SemanticCheckException(msg, e);
+    if (IPUtils.compare(address, range.getLower()) < 0) {
+      return ExprValueUtils.LITERAL_FALSE;
     }
 
-    // Address and range must use the same IP version (IPv4 or IPv6).
-    if (address.isIPv4() ^ range.isIPv4()) {
-      String msg =
-          String.format(
-              "IP address '%s' and CIDR IP address range '%s' are not compatible. Both must be"
-                  + " either IPv4 or IPv6.",
-              addressString, rangeString);
-      throw new SemanticCheckException(msg);
+    if (IPUtils.compare(address, range.getUpper()) > 0) {
+      return ExprValueUtils.LITERAL_FALSE;
     }
 
-    return ExprValueUtils.booleanValue(range.contains(address));
+    return ExprValueUtils.LITERAL_TRUE;
```

Comment:
nit: 
```
return (IPUtils.compare(address, range.getLower()) < 0) || (IPUtils.compare(address, range.getUpper()) > 0) ? ExprValueUtils.LITERAL_FALSE : ExprValueUtils.LITERAL_TRUE;
```

Reply:
Funny you should comment on this: I actually wrote out a comment here, which I apparently forgot to ever actually post 🤦. It went something like:

>  I know that this could be combined into a single statement, but I generally like breaking this out, even if it's more verbose; I tend to find it slightly more readable/understandable. Not sure if there is a style that is more common on this project or in Java? That being said, happy to change it if you think it's better off being more compact

Long way of saying that I've gone ahead and updated it as suggested 😆

Reply:
Oh lol!  Generally, I don't mind verbosity.  But this is too many extra lines of code for my taste.  Generally, I don't have a cut off, but merging 5 lines into 1 sounds like a good tradeoff.